addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -123,211 +144,173 @@ public void run() {
+            logger.debug("{}:    Port Telnet         {}", getLogIdentifier(), config.getTelnetPort());
+            logger.debug("{}:    Portal Path         {}", getLogIdentifier(), config.getServerBasePath());
+            logger.debug("{}:    API Path            {}", getLogIdentifier(), config.getServerApiPath());
+            logger.debug("{}:    Refresh interval:   {}", getLogIdentifier(), config.getRefreshIntervalNormal());
+            logger.debug("{}:    Low  prio. refresh: {}", getLogIdentifier(), config.getRefreshIntervalLow());
+            logger.debug("{}:    Autodiscovery:      {}", getLogIdentifier(), config.getAutodiscoverThings());
+
             startWatchDogTask();
-            isInitialized = true;
-        }
-    }
+            initialized = true;
+            return;
+        } catch (Exception ex) {
+            if (zoneMinderConnection == null) {
+                logger.error(
+                        "{}: 'ZoneMinderServerBridgeHandler' general configuration error occurred. Failed to initialize.",
+                        getLogIdentifier(), ex);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            } else {
+                logger.error("{}: 'ZoneMinderServerBridgeHandler' failed to initialize", getLogIdentifier(), ex);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
+            }
 
-    /**
-     * Method to find the lowest possible refresh rate (based on configuration)
-     *
-     * @param refreshRate
-     * @return
-     */
-    protected int calculateCommonRefreshFrequency(int refreshRate) {
-        // Check if 30, 15, 10 or 5 seconds is possible
-        if ((refreshRate % 30) == 0) {
-            return 30;
-        } else if ((refreshRate % 15) == 0) {
-            return 15;
-        } else if ((refreshRate % 10) == 0) {
-            return 10;
-        } else if ((refreshRate % 5) == 0) {
-            return 5;
         }
 
-        // Hmm, didn't find a obvious shared value. Run every second...
-        return 1;
+        initialized = false;
+
+    }
 
+    protected IZoneMinderConnectionHandler aquireSession() {
+        return zoneMinderConnection;
+    }
+
+    protected void releaseSession() {
     }
 
     protected void startWatchDogTask() {
-        taskWatchDog = startTask(watchDogRunnable, 0, 15, TimeUnit.SECONDS);
+        taskWatchDog = startTask(watchDogRunnable, 5, 1, TimeUnit.SECONDS);
     }
 
     protected void stopWatchDogTask() {
         stopTask(taskWatchDog);
         taskWatchDog = null;
     }
 
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        try {
+            setConnected(false);
+            zoneMinderConnection = null;
+            setConnectionStatus(ZoneMinderConnectionStatus.UNINITIALIZED);
+        } catch (IllegalArgumentException | GeneralSecurityException | IOException | ZoneMinderUrlNotFoundException e) {
```

Comment:
Can this small block of code really throw all of this on us?

Reply:
Unfortunately yes. The problem is that the ZoneMinder API seems to always return a HTTP Code 401 , so I have to guess what ever the correct reason is. There is also a of combinations that can cause problems. 
* Authentication can be disabled, but API disabled => return code 401 , 
* Authentication can be enabled, wrong credentials  => return code 401 , 
* Authentication can be enabled, insufficent user rights in ZM => return code 401 , 
Just to mention a few, also things seems to differ in different ZoneMinder versions. I can't really see if authentication is enabled or disabled, just have to try and see what happends...

SInce the Jetty Client also uses exceptions, I did continue on that  road. To identify the correct combination I simply use trial and error. It might be possible to rewrite code to minimize usage of different exceptions. But it really requires intensive testing. 
This would be quite a change, so I would really avoid this for now.